### PR TITLE
instancetype: Improve admission conflict message

### PR DESCRIPTION
**Pull request by lyarwood**
/area instancetype

**What this PR does / why we need it**:

This should really contain the field causing the conflict to avoid the reader having to look around further for this information.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [x] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by iholder101** [comment]
Thanks @lyarwood!
I have tested it locally and it works great. Now the message is much much better. Thank you.
Is there a reason this PR is still a draft?

/approve

**Comment by lyarwood** [comment]
> Thanks @lyarwood! I have tested it locally and it works great. Now the message is much much better. Thank you. Is there a reason this PR is still a draft?

/cc @0xFelix 

Nope was just posted in a rush, I've marked it as ready to review now thanks.

**Review by 0xFelix — APPROVED** [review]
/lgtm

Thanks!

File: tests/instancetype_test.go [hunk 1]
```diff
@@ -468,18 +471,21 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 
 			cause0 := apiStatus.Status().Details.Causes[0]
 			Expect(cause0.Type).To(Equal(metav1.CauseTypeFieldValueInvalid))
-			Expect(cause0.Message).To(Equal("VM field conflicts with selected Instancetype"))
-			Expect(cause0.Field).To(Equal("spec.template.spec.domain.cpu.sockets"))
+			cpuSocketsField := "spec.template.spec.domain.cpu.sockets"
```
Comment (0xFelix) [inline tests/instancetype_test.go:474]: Make it const?

File: tests/instancetype_test.go [hunk 2]
```diff
@@ -468,18 +471,21 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
...
+			Expect(cause0.Message).To(Equal(fmt.Sprintf(instancetypepkg.VMFieldConflictErrorFmt, cpuSocketsField)))
+			Expect(cause0.Field).To(Equal(cpuSocketsField))
 
 			cause1 := apiStatus.Status().Details.Causes[1]
 			Expect(cause1.Type).To(Equal(metav1.CauseTypeFieldValueInvalid))
-			Expect(cause1.Message).To(Equal("VM field conflicts with selected Instancetype"))
-			Expect(cause1.Field).To(Equal("spec.template.spec.domain.cpu.cores"))
+			cpuCoresField := "spec.template.spec.domain.cpu.cores"
```
Comment (0xFelix) [inline tests/instancetype_test.go:480]: Same

File: tests/instancetype_test.go [hunk 3]
```diff
@@ -468,18 +471,21 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
...
+			Expect(cause1.Message).To(Equal(fmt.Sprintf(instancetypepkg.VMFieldConflictErrorFmt, cpuCoresField)))
+			Expect(cause1.Field).To(Equal(cpuCoresField))
 
 			cause2 := apiStatus.Status().Details.Causes[2]
 			Expect(cause2.Type).To(Equal(metav1.CauseTypeFieldValueInvalid))
-			Expect(cause2.Message).To(Equal("VM field conflicts with selected Instancetype"))
-			Expect(cause2.Field).To(Equal("spec.template.spec.domain.cpu.threads"))
+			cpuThreadsField := "spec.template.spec.domain.cpu.threads"
```
Comment (0xFelix) [inline tests/instancetype_test.go:486]: Same

File: tests/instancetype_test.go [hunk 4]
```diff
@@ -32,6 +33,8 @@ import (
 	"kubevirt.io/client-go/kubecli"
 	cdiv1beta1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
 
+	instancetypepkg "kubevirt.io/kubevirt/pkg/instancetype"
+
```
Comment (0xFelix) [inline tests/instancetype_test.go:37]: nit: newline

**Review by 0xFelix — APPROVED** [review]
/lgtm

Thanks!

**Pull request merged**